### PR TITLE
added functionalities for extra items to populate around nodes

### PR DESCRIPTION
**Pull request by ramizbalayil**

### CODE REVIEW

File: layers/extras/GiftingLevel.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/* global cc, adv_map: true
```
Comment (rampr) [inline layers/extras/GiftingLevel.js]: We can't have generic names for the extras ?

**Review by nasimafeef — APPROVED** [review]

File: layers/extras/GiftingLevel.js (hunk 1 above)
Comment (ramizbalayil) [reply layers/extras/GiftingLevel.js]: There is a class called NodeExtra from which we extend for both Gifting and Bonus Level. But as discussed I will find a way to handle all animations and functionalities from game

File: layers/TileLayer.js [hunk 2]
```diff
@@ -20,6 +22,10 @@ adv_map.layers.TileLayer = ccui.Widget.extend({
 
     this._super();
     this.navigator_data = [];
+    this.extra_data = {
+      BonusLevel: false,
+      GiftingLevel: false
```
Comment (vivekb91) [inline layers/TileLayer.js]: Ideally these things shouldn't here. I think this we can avoid

File: layers/TileLayer.js [hunk 3]
```diff
@@ -165,6 +177,33 @@ adv_map.layers.TileLayer = ccui.Widget.extend({
...
+    if (node) {
+      extra = new this.node_settings.extras[id]({
+        ms: this.ms_number,
+        size: node.getContentSize()
+      });
+      node.addChild(extra);
+      extra.setTouchEnabled(true);
+      extra.addTouchEventListener(function (target, type) {
+        if (type === ccui.Widget.TOUCH_ENDED) {
+          cc.eventManager.dispatchCustomEvent('map_extra', {
```
Comment (vivekb91) [inline layers/TileLayer.js]: tag_selected

File: index.js [hunk 4]
```diff
@@ -143,6 +143,22 @@ adv_map.AdventureMapLayer = cc.Layer.extend({
...
+    'use strict';
+
+    var layers = this.scrollable_map.map_children,
+      tile_layers = [],
+      i;
+
+    for (i = layers.length - 1; i >= 0; i--) {
+      if (layers[i].ms_number > 0) {
+        tile_layers.push(layers[i]);
+      }
```
Comment (vivekb91) [inline index.js]: use _ function, I think _. filter will work

File: layers/TileLayer.js (hunk 2 above)
Comment (ramizbalayil) [reply layers/TileLayer.js]: done

File: layers/TileLayer.js (hunk 3 above)
Comment (ramizbalayil) [reply layers/TileLayer.js]: done

File: index.js (hunk 4 above)
Comment (ramizbalayil) [reply index.js]: done

**Review by vivekb91 — APPROVED** [review]

File: layers/TileLayer.js [hunk 5]
```diff
@@ -165,6 +174,33 @@ adv_map.layers.TileLayer = ccui.Widget.extend({
...
+        if (type === ccui.Widget.TOUCH_ENDED) {
+          cc.eventManager.dispatchCustomEvent('tag_selected', {
+            tag: extra.node_tag,
+            ms: extra.getParent().milestone
+          });
+        }
+      }, extra);
+      this.extra_data[id] = true;
+    } else {
+      this.extra_data[id] = true;
```
Comment (rampr) [inline layers/TileLayer.js]: this statement is there in both if and else. Can be moved out of the conditions

**Review by nasimafeef — APPROVED** [review]

**Review by nasimafeef — APPROVED** [review]

File: layers/TileLayer.js (hunk 5 above)
Comment (ramizbalayil) [reply layers/TileLayer.js]: done

**Pull request merged**